plugins/MosaicCamera/src/MosaicCamera.cpp

```diff
@@ -0,0 +1,560 @@
+{
+    if (cameras.contains(cameraName))
+    {
+        cameras[cameraName].visible = visible;
+        if(configDialog->visible()) {
+            if (configDialog->getCurrentCameraName() == cameraName) {
+                configDialog->setVisibility(visible);
+            }
+        }
+    }
+}
+
+void MosaicCamera::setPosition(const QString& cameraName, double ra, double dec, double rotation)
+{
+    setRA(cameraName, ra);
+    setDec(cameraName, dec);
+    setRotation(cameraName, rotation);
+}
+
+void MosaicCamera::setRADecToView()
+{
+	Vec3d current = GETSTELMODULE(StelMovementMgr)->getViewDirectionJ2000();
+	double ra, dec;
+	StelUtils::rectToSphe(&ra, &dec, current);
+    // convert to degrees from radians
+	ra = std::fmod((ra*180/M_PI)+360., 360.);
+	dec = dec*180/M_PI;
+    setCurrentRA(ra);
+    setCurrentDec(dec);
+}
+
+void MosaicCamera::incrementRotation()
+{
+    double rotation = getRotation(currentCamera);
+    rotation += 5.0;
+    setRotation(currentCamera, rotation);
+}
+
+void MosaicCamera::decrementRotation()
+{
+    double rotation = getRotation(currentCamera);
+    rotation -= 5.0;
+    setRotation(currentCamera, rotation);
+}
+
+void MosaicCamera::nextCamera()
+{
+    int index = cameraOrder.indexOf(currentCamera);
+    index = (index + 1) % cameraOrder.size();
+    setCurrentCamera(cameraOrder[index]);
+}
+
+void MosaicCamera::previousCamera()
+{
+    int index = cameraOrder.indexOf(currentCamera);
+    index = (index - 1 + cameraOrder.size()) % cameraOrder.size();
+    setCurrentCamera(cameraOrder[index]);
+}
+
+double MosaicCamera::getRA(const QString& cameraName) const
+{
+    return cameras.value(cameraName).ra;
+}
+
+double MosaicCamera::getDec(const QString& cameraName) const
+{
+    return cameras.value(cameraName).dec;
+}
+
+double MosaicCamera::getRotation(const QString& cameraName) const
+{
+    return cameras.value(cameraName).rotation;
+}
+
+bool MosaicCamera::getVisibility(const QString& cameraName) const
+{
+    return cameras.value(cameraName).visible;
+}
+
+void MosaicCamera::setCurrentCamera(const QString& cameraName)
+{
+    if (cameras.contains(cameraName))
+    {
+        currentCamera = cameraName;
+    }
+}
+
+void MosaicCamera::setCurrentRA(double ra)
+{
+    setRA(currentCamera, ra);
+}
+
+void MosaicCamera::setCurrentDec(double dec)
+{
+    setDec(currentCamera, dec);
+}
+
+void MosaicCamera::setCurrentRotation(double rotation)
+{
+    setRotation(currentCamera, rotation);
+}
+
+void MosaicCamera::setCurrentVisibility(bool visible)
+{
+    setVisibility(currentCamera, visible);
+}
+
+void MosaicCamera::draw(StelCore* core)
+{
+	if (!isEnabled())
+		return;
+
+    const StelProjectorP prj = core->getProjection(StelCore::FrameJ2000);
+    StelPainter painter(prj);
+
+    Vec3d startPoint, endPoint;
+
+    for (const auto& camera : cameras)
+    {
+        if (!camera.visible)
+            continue;
+
+        double alpha = camera.ra / 57.29577951308232 + 1.5707963267948966;
```

Comment:
Please use the usual named constants from StelUtils.hpp. M_PI_2 ~ 1.57..., M_180_PI ~ 57.29...